Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/Implementation/Tracing/HeartbeatTests.cs

```diff
@@ -0,0 +1,415 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.Mocks;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class HealthHeartbeatTests
+    {
+        [TestMethod]
+        public void InitializeHealthHeartbeatDoesntThrow()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatTwiceDoesntFail()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: null);
+                hbeat.Initialize(configuration: null);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatDefaultsAreSetProperly()
+        {
+            using (var hbeat = new HealthHeartbeatProviderMock())
```

Comment:
Should this be testing the actual HeartbeatProvider() ?

Reply:
Yes. I've altered the class definition slightly to allow for testing to occur a bit more simply. I was extending it to gain access to the private members of the class, and to disable the timer during unit tests originally. However, I like it better that we test the class directly so I've done away with the mock completely now.